python/ql/lib/semmle/python/AstExtended.qll

```diff
@@ -154,6 +154,24 @@ class StringPart extends StringPart_, AstNode {
   override string toString() { result = StringPart_.super.toString() }
 
   override Location getLocation() { result = StringPart_.super.getLocation() }
+
+  /** Holds if the content of string `StringPart` is surrounded by `prefix` and `quote`. */
+  predicate context(string prefix, string quote) {
```

Comment:
It seems like callers of this predicate are really only interested in the length of the prefix and the quote; perhaps it makes sense to just expose that?

Reply:
I did consider that; it might also be easier on the string pool. It just did not feel as generally useful, but perhaps that is a silly concern. (And given that the prefix and the quote are part of the string, it is actually the same information.)